giraffe/src/components/AnnotationLine.tsx

```diff
@@ -166,6 +191,12 @@ export const AnnotationLine: FunctionComponent<AnnotationLineProps> = props => {
           style: {cursor: 'pointer'},
           id: props.id,
           className: 'giraffe-annotation-click-target',
+          onMouseEnter: () => {
+            handleMouseEnter()
+          },
+          onMouseLeave: () => {
+            handleMouseLeave()
+          },
```

Comment:
Since there is nothing unique about each `onMouseEnter` and `onMouseLeave` function, there is no need to create a separate copy of each function every time `makeRangeRectangle` is called. Ultimately, you just want to call `onHover(true)` or `onHover(false)`

I would remove the unnecessary wrapping
```
      onMouseEnter: handleMouseEnter,
      onMouseLeave: handleMouseLeave,
```